src/super_gradients/training/metrics/pose_estimation_metrics.py

```diff
@@ -0,0 +1,286 @@
+import itertools
+from typing import Dict, Union, List, Optional, Tuple, Callable, Iterable, Any
+
+import numpy as np
+import torch
+from torch import Tensor
+from torchmetrics import Metric
+
+import super_gradients
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.common.registry.registry import register_metric
+from super_gradients.training.metrics.pose_estimation_utils import compute_img_keypoint_matching, compute_visible_bbox_xywh
+from super_gradients.training.utils.detection_utils import compute_detection_metrics_per_cls
+
+logger = get_logger(__name__)
+
+__all__ = ["PoseEstimationMetrics"]
+
+
+@register_metric("PoseEstimationMetrics")
+class PoseEstimationMetrics(Metric):
+    """
+    Implementation of COCO Keypoint evaluation metric
+    """
+
+    def __init__(
+        self,
+        post_prediction_callback: Callable[[Any], Tuple[Tensor, Tensor]],
+        num_joints: int,
+        max_objects_per_image: int = 20,
+        oks_sigmas: Optional[Iterable] = None,
+        iou_thresholds: Optional[Iterable] = None,
+        recall_thresholds: Optional[Iterable] = None,
+        iou_thresholds_to_report: Optional[Iterable] = None,
+    ):
+        """
+        Compute the AP & AR metrics for pose estimation. By default, this class returns only AP and AR values.
+        If you need to get additional metrics (AP at specific threshold), pass these thresholds via `iou_thresholds_to_report` argument.
+
+        :param post_prediction_callback:
+        :param num_joints:
+        :param oks_sigmas: OKS sigma factor for custom keypoint detection dataset.
+                           If None, then metric will use default OKS from COCO and expect num_joints to be equal 17
+        :param recall_thresholds: List of recall thresholds to compute AP.
+                                  If None, then will use default 101 recall thresholds from COCO in range [0..1]
+        :param iou_thresholds: List of IoU thresholds to use. If None, then COCO version of IoU will be used (0.5 ... 0.95)
+        :param: iou_thresholds_to_report: List of IoU thresholds to return in metric.
```

Comment:
`max_objects_per_image` is not described

Reply:
Nice catch!